tripal/tripal
# Backport of File Usage and Expiration.

Issue #417

## Type(s) of Change(s)

- [X] Backport

## Description
Tripal v3 has a file quota and expiration system.  This is a backport of that code to Tripal v2.  It is needed for compatibility of the Tripal Galaxy module with both Tv2 and Tv3.

## Testing?




The easiest way to test is to enable the Tripal Galaxy module and do the following:
1)  First pull the branch and run a ```drush updatedb``` to get new tables.
2)  Create a new Galaxy workflow submission (via the webform) that requires file uploads.
3) Upload a file larger than 64MB as this is the default quota size. You should get an error message that the file exceeds the user quota and cannot be uploaded.
4)  Go the Tripal > User File Management  and change the default quota to something large enough for the file upload.
5) Return to the workflow and upload the file again. It should now upload.
6) Go back to the Tripal > User File Management and set the default quota back to something very small.
7) Now return to the Galaxy workflow and try to upload a new file larger than the quota limit. It should fail.
8) Repeat steps 3-7 but this time test that a user specific quota works.  There are options on the quota management page for setting a user quota.  You should not be able to upload a file for a specific user if it exceeds the quota.
9) To to your user account page and you should see a new 'Uploads' tab.  All of the files you uploaded should be there and you should be able to download and delete them.

Because this is a backport and does not represent any new code that isn't already in use for Tv3  I think we only need one reviewer for this.  @MingChen0919 and @NicHerndon can you provide a review for me as you both use the Galaxy module?  I just need one review, so who ever can do it first. @NicHerndon if you don't have a Tv2 site to test with, then no worries.

## Screenshots (if appropriate):

## Additional Notes (if any):
